src/agora/dobeamweights.h

```diff
@@ -51,16 +51,18 @@ class DoBeamWeights : public Doer {
   EventData Launch(size_t tag) override;
 
  private:
-  void ComputePartialCsiBeams(size_t tag);
+  static inline void PartialTransposeGather(size_t cur_sc_id, float* src,
```

Comment:
Not sure why these are static?  probably doesn't hurt but I would hid them in the .cc file as static (not part of the class) or you can make them private member functions (non static).